tests/data_files/Makefile

```diff
@@ -1087,6 +1087,43 @@ cert_md5.crt: cert_md5.csr
 	$(MBEDTLS_CERT_WRITE) request_file=$< serial=6 issuer_crt=$(test_ca_crt) issuer_key=$(test_ca_key_file_rsa) issuer_pwd=$(test_ca_pwd_rsa) not_before=20000101121212 not_after=20300101121212 md=MD5 version=3 output_file=$@
 all_final += cert_md5.crt
 
+# TLSv1.3 test certificates
+ecdsa_secp256r1_sha256.key: ec_256_prv.pem
+	ln -s $< $@
+ecdsa_secp256r1_sha256.csr: ecdsa_secp256r1_sha256.key
```

Comment:
I think that this name together with ecdsa_secp384r1_sha384 and ecdsa_secp521r1_sha512 are misleading as they seem to imply that there is link between the number of bits of the key certified by the certificate and the number of bits of the hash function used to sign the certificate. And there is no such link.

What the number of bits of the hash function is more related to is the number of bits of the key used to sign the certificate thus here the private key $(test_ca_key_file_ec) == test-ca2.key. The size of this private key is 384 bits thus the best hash (maximum possible security level with minimum computation for that security level) to use with it is sha384 and I think it would be better (and probably more realistic) to sign all three certificates using ecdsa with sha384.

Then I think for the name of the key/csr/crt files I would just go with "ecdsa_secp256/384/521r1". Those are files associated to a 256/384/521 bits EC key that can be used for ecdsa signature. Then any hash function can be used for the ecdsa signature but in the context of TLS 1.3 it is going to be used only with sha256/sha384/sha512.

Reply:
Please double check. 
It is in next commit of `fix pylint fail` .